javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -124,9 +124,11 @@ private static int treeSize(int branchCount, int depthShift) {
         } else if (n >= length()) {
             return empty();
         } else {
-            final NodeModifier drop = Arrays::copyDrop;
-            final Object[] root = modifyLeaf(array, depthShift, offset + n, drop, drop);
-            return new BitMappedTrie<>(root, offset + n, length() - n, depthShift);
+            final int index = offset + n;
```

Comment:
BMT nodes are dropped, but leafs aren't: they're masked by `offset` and `length`.
The first array was always prepended by `null` anyway, so only the last node might be bigger this way by at most `31` elements. However, since it's not copied at all, it will usually save a lot more memory.